core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala

```diff
@@ -614,14 +614,30 @@ class SnowflakeExpressionBuilder(override val vc: SnowflakeVisitorCoordinator)
   }
 
   override def visitPredIn(ctx: PredInContext): ir.Expression = {
-    val in = if (ctx.subquery() != null) {
+    // Handling the left side of the IN clause (either an exprList or single expression)
+    val left = if (ctx.exprList() != null) {
+      // Map over the expressions in exprList and collect them as a list of ir.Expression
+      // need to figure out why exprList().expr() is not working
+      ctx.exprList().asScala.flatMap(_.expr().asScala.map(_.accept(this)))
+    } else {
+      // Handle case where there's no exprList, and we have a single expression
+      Seq(ctx.exprList().asScala.map(_.accept(this)))
+    }
+    // Handling the right side of the IN clause (either a subquery or exprList)
+    val right = if (ctx.subquery() != null) {
       // In the result of a sub query
-      ir.In(ctx.expression().accept(this), Seq(ir.ScalarSubquery(ctx.subquery().accept(vc.relationBuilder))))
+      Seq(ir.ScalarSubquery(ctx.subquery().accept(vc.relationBuilder)))
     } else {
-      // In a list of expressions
-      ir.In(ctx.expression().accept(this), ctx.exprList().expr().asScala.map(_.accept(this)))
+      Seq(ctx.exprList().asScala.head.accept(this))
+    }
+    left.size match {
+      case 1 =>
+        val in = ir.In(left.head, right) // If there's only one expression on the left side
```

Comment:
Revert to the existing code for `visitPredIn` and implement `visitPredExprListIn` if ther is overlap then share the overlap with a buildXXX def to handle the common features so that the visitors are actually quite simple